src/public/js/util.js

```diff
@@ -146,6 +146,9 @@ var UTIL = (function() {
   }
 
   function polynomialDiv(dividend, divisor, n, k) {
+    // Make a copy of dividend to avoid modifying input
+    dividend = dividend.slice(0);
```

Comment:
Good catch, I did some reading on JS function arguments specifically for this function call and I read that parameters are passed by value and left it at that. Of course in JS the arrays are objects and not built in types so passing an array passes a value of the ref of the array so in this case I was being silly! This seems like a good work around. Another option would be to copy msg in to codedmsg before the function call. The way you have it hear is better code in my opinion, although the other way would be somewhat more optimized as it would remove the extra copy.